### ISSUE TITLE
 PoissinPointProcess does not compute prior density for position and attributes

### ISSUE THREAD

**Issue by mikkelbue** [mikkelbue/tinyDA#53]
The [`logpmf`](https://github.com/mikkelbue/tinyDA/blob/3e2660c4f295aadf04efb4fad3bf3abe7f65e0d2/tinyDA/distributions.py#L164) method of `PoissonPointProcess` currectly only computes the Poisson density and does not check whether the points are all within the domain. It should also check that all points are within the domain using the `domain_dist`, and return `-inf` if they aren't and simply the Poisson density if they are.

Additionally, the point attributes should all be uniform (to ensure detailed balance). This could be asserted by the `__init__` method. Then, the `logpdf` method could check that the attributes are all within the bounds in a fashion similar to the domain distribution.

In short: `logpdf` of `PoissonPointProcess` must return `self.pois.logpmf(len(x))` if all the point locations and attributes are within their uniform bounds, and `-inf` if any of them isn't.
